### PR TITLE
PR/Release: v0.12.0 [WIP]

### LINKED ISSUE #62: Type hinting throughout the code base
Add type hints to later integrate into our documentation.

### LINKED ISSUE #64: Enhancement: Add CI testing for Python 3.9
**Summary**
For future enhancements including typing checks, we should include CI for Python 3.9.

**Objectives**

Eventually take advantage of Python 3.9 typing features

**Proposal**

Modify python-package.yml to include Python 3.9.

Implemented in: TBD

### LINKED ISSUE #66: Sphinx documentation with RTDs
We are ready to provide documentation through ReadTheDocs.

This [tutorial](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/sphinx-quickstart.html) is a good starter. A couple of key points:
1. Docs will be in a `docs/` folder relative to the repo parent
2. We will keep source and build separate, so answer yes to that question
3. We will build with python 3.7
4. We will transfer README.md over and have separate sections (separate .rst files)
5. We will use `sphinx_rtd_theme` HTML theme. You will need to include to include the `sphinx_rtd_theme` extension
 
Action item:
- [x] Clean up all docstrings
- [x] Include requirements.txt under `docs/` for Sphinx compilation
- [ ] Include a GitHub Action script
- [x] Add shields.io badges. See this [example](https://github.com/astrochun/Metallicity_Stack_Commons)

### PR DESCRIPTION
**Pull request by astrochun**
**Description**

This is a full minor release including sphinx documentation and type hingintg

Closes #62 
Closes #64 
Closes #66 

**Bump version**
v0.11.1 -> v0.12.0

- [x] README.md, [installation instructions](../../README.md#installation-instructions)
- [x] [`setup.py`](../../setup.py)
- [x] [`requiam_csv/__init__.py`](../../requiam_csv/__init__.py)
- [x] ReadTheDocs integration with repository

*Hyperlink documentation for Sphinx*
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks

### CODE REVIEW

**Review by damian-romero — APPROVED** [review]
Changes look good except for broken links in:
- license.rst
- overview.rst (requires a reference to execution.rst)

I'll make these changes and push them to release/v0.12.0 along with the bump version.

**Comment by damian-romero** [comment]
@astrochun I believe you said in today's live coding that some links were not working for you. I'm not sure if the ones I fixed were the same ones you were having issues with.

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+sphinx>=3.0
+sphinx-rtd-theme==0.5.1
+sphinx-autodoc-typehints==1.11.1
+numpy==1.18.0
+pandas==0.25.3
```
Comment (astrochun) [inline docs/requirements.txt]:
```suggestion
numpy==1.19.5
pandas==1.2.1
```

**Review by astrochun — COMMENTED** [review]
Ready to merge with tags pushed.

Note: Update docs/requirements.txt for consistency

**Pull request merged**